File: api/app/services/branch_service.py
from app.repositories.branch_repository import (
    BranchRepository
)


class BranchService:

    @staticmethod
    def get_all_branches():
        return BranchRepository.find_all()

    @staticmethod
    def get_branch_by_id(
        branch_id: int
    ):
        return BranchRepository.find_by_id(
            branch_id
        )

    @staticmethod
    def get_tenant_branches(
        tenant_id: int
    ):
        return BranchRepository.find_by_tenant_id(
            tenant_id
        )

    @staticmethod
    def create_branch(payload):
        return BranchRepository.create(
            payload
        )

    @staticmethod
    def update_branch(
        branch_id: int,
        payload
    ):
        return BranchRepository.update(
            branch_id,
            payload
        )

    @staticmethod
    def delete_branch(
        branch_id: int
    ):
        return BranchRepository.delete(
            branch_id
        )

File: web/src/services/branch-service.ts
import { api } from "./api";
import type { Branch, BranchCreatePayload, BranchUpdatePayload } from "@/types/branch";

export async function getBranches() {
  const response = await api.get("/branches");
  return response.data.data as Branch[];
}

export async function getBranch(branchId: number) {
  const response = await api.get(`/branches/${branchId}`);
  return response.data.data as Branch;
}

export async function getTenantBranches(tenantId: number) {
  const response = await api.get(`/branches/tenant/${tenantId}`);
  return response.data.data as Branch[];
}

export async function createBranch(payload: BranchCreatePayload) {
  const response = await api.post("/branches", payload);
  return response.data.data as Branch;
}

export async function updateBranch(branchId: number, payload: BranchUpdatePayload) {
  const response = await api.put(`/branches/${branchId}`, payload);
  return response.data.data as Branch;
}

export async function deleteBranch(branchId: number) {
  const response = await api.delete(`/branches/${branchId}`);
  return response.data;
}
